### PR TITLE
[SMALLFIX] Clean up test configuration usage in PermissionCheckTest

### PR DESCRIPTION
**Pull request by aaudiber**
Avoid manual configuration modification by using `AlluxioTestDirectory` instead of `mTestFolder`

### CODE REVIEW

File: core/server/master/src/test/java/alluxio/master/file/PermissionCheckTest.java [hunk 1]
```diff
@@ -113,9 +112,13 @@
...
   @Rule
-  public ConfigurationRule mConfiguration = new ConfigurationRule(ImmutableMap
-      .of(PropertyKey.SECURITY_GROUP_MAPPING_CLASS, FakeUserGroupsMapping.class.getName(),
-          PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP, TEST_SUPER_GROUP));
+  public ConfigurationRule mConfiguration =
+      new ConfigurationRule(new ImmutableMap.Builder<PropertyKey, String>()
+          .put(PropertyKey.SECURITY_GROUP_MAPPING_CLASS, FakeUserGroupsMapping.class.getName())
+          .put(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP, TEST_SUPER_GROUP)
+          .put(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS, AlluxioTestDirectory
+              .createTemporaryDirectory("PermissionCheckTest").getAbsolutePath())
```
Comment (apc999) [inline core/server/master/src/test/java/alluxio/master/file/PermissionCheckTest.java]: who will be responsible to remove this temp dir?

Comment (aaudiber) [inline core/server/master/src/test/java/alluxio/master/file/PermissionCheckTest.java]: The `AlluxioTestDirectory` class itself is responsible for cleaning up the directories it creates. It does this by registering a JVM exit hook, as well as cleaning up previous `AlluxioTestDirectory`s older than 1 hour.

Comment (apc999) [inline core/server/master/src/test/java/alluxio/master/file/PermissionCheckTest.java]: ack. good to know

**Comment by aaudiber** [comment]
@calvinjia I looked around and found another few tests doing the same thing. Addressed in https://github.com/Alluxio/alluxio/pull/5775. There are also many more tests that use `Configuration.set` that could be changed to use a `ConfigurationRule`. I've created a JIRA ticket for this: https://alluxio.atlassian.net/browse/ALLUXIO-2962

**Pull request merged**